ci-operator/step-registry/ipi/conf/azure/default/ipi-conf-azure-default-commands.sh

```diff
@@ -0,0 +1,116 @@
+# release:initial image, once that, use 'oc get istag release:inital' to workaround it.
+echo "RELEASE_IMAGE_INITIAL: ${RELEASE_IMAGE_INITIAL:-}"
+if [[ -n ${RELEASE_IMAGE_INITIAL:-} ]]; then
+    tmp_release_image_initial=${RELEASE_IMAGE_INITIAL}
+    echo "Getting inital release image from RELEASE_IMAGE_INITIAL..."
+elif oc get istag "release:initial" -n ${NAMESPACE} &>/dev/null; then
+    tmp_release_image_initial=$(oc -n ${NAMESPACE} get istag "release:initial" -o jsonpath='{.tag.from.name}')
+    echo "Getting inital release image from build farm imagestream: ${tmp_release_image_initial}"
+fi
+# For some ci upgrade job (stable N -> nightly N+1), RELEASE_IMAGE_INITIAL and 
+# RELEASE_IMAGE_LATEST are pointed to different imgaes, RELEASE_IMAGE_INITIAL has 
+# higher priority than RELEASE_IMAGE_LATEST
+TESTING_RELEASE_IMAGE=""
+if [[ -n ${tmp_release_image_initial:-} ]]; then
+    TESTING_RELEASE_IMAGE=${tmp_release_image_initial}
+else
+    TESTING_RELEASE_IMAGE=${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}
+fi
+echo "TESTING_RELEASE_IMAGE: ${TESTING_RELEASE_IMAGE}"
+
+dir=$(mktemp -d)
+pushd "${dir}"
+cp ${CLUSTER_PROFILE_DIR}/pull-secret pull-secret
+oc registry login --to pull-secret
+version=$(oc adm release info --registry-config pull-secret ${TESTING_RELEASE_IMAGE} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+major_version=$( echo "${version}" | awk --field-separator=. '{print $1}' )
+minor_version=$( echo "${version}" | awk --field-separator=. '{print $2}' )
+echo "get ocp version: ${version}"
+rm pull-secret
+popd
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+REGION="${LEASED_RESOURCE}"
+echo "Azure region: ${REGION}"
+
+workers=${COMPUTE_NODE_REPLICAS:-3}
+if [ "${COMPUTE_NODE_REPLICAS}" -le 0 ]; then
+  workers=0
+fi
+
+cat >> "${CONFIG}" << EOF
+baseDomain: ${BASE_DOMAIN}
+platform:
+  azure:
+    region: ${REGION}
+controlPlane:
+  architecture: ${OCP_ARCH}
+  name: master
+  platform: {}
+compute:
+- architecture: ${OCP_ARCH}
+  name: worker
+  replicas: ${workers}
+  platform: {}
+EOF
+
+if [ -z "${OUTBOUND_TYPE}" ]; then
+  echo "Outbound Type is not defined"
+else
+  if [ X"${OUTBOUND_TYPE}" == X"UserDefinedRouting" ] || [ X"${OUTBOUND_TYPE}" == X"NatGateway" ]; then
+    echo "Writing 'outboundType: ${OUTBOUND_TYPE}' to install-config"
+    PATCH="${SHARED_DIR}/install-config-outboundType.yaml.patch"
+    cat > "${PATCH}" << EOF
+platform:
+  azure:
+    outboundType: ${OUTBOUND_TYPE}
+EOF
+    yq-go m -x -i "${CONFIG}" "${PATCH}"
+  else
+    echo "${OUTBOUND_TYPE} is not supported yet" || exit 1
+  fi
+fi
+
+# User tags went GA in 4.14. In 4.14+, tag resources with an expiration date to facilitate cleanup.
+if (( minor_version > 13 && major_version == 4 )); then
+  expiration_date=$(date -d '8 hours' --iso=minutes --utc)
+  printf 'Setting user tag expirationDate: %s\n' "${expiration_date}"
+  yq-go write -i "${CONFIG}" "platform.azure.userTags.expiration_date" "${expiration_date}"
```

Comment:
Shall we drop `userTags.expiration_date` setting here to get a pretty minimal install-config.yaml ? or we can skip the version check, directly set `platform.azure.userTags.expiration_date` in install-config.yaml, because the testing is only run in 4.16+

Reply:
dropped